File: miscellaneous/graph-dfs/20230726/graph-dfs-iter.php
<?php

function dfs(array $adjList, int $start): array
{
    $stack = [$start];
    $seen = [$start];
    $values = [];

    while ($stack) {
        $current = array_pop($stack);
        $values[] = $current;
        $connections = $adjList[$current];

        foreach ($connections as $connection) {
            if (!in_array($connection, $seen)) {
                $seen[] = $connection;
                $stack[] = $connection;
            }
        }
    }

    return $values;
}

File: miscellaneous/graph-dfs/20230726/graph_dfs_iter.py
from typing import List, Tuple


def dfs(adj_list: List[int], start: int) -> Tuple[int]:
    values = []
    stack = [start]
    seen = [start]

    while stack:
        current = stack.pop()
        values.append(current)
        connections = adj_list[current]

        for connection in connections:
            if connection not in seen:
                stack.append(connection)
                seen.append(connection)

    return tuple(values)